ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,199 @@
+                                " component!"
+                            )
+
+                    else:
+                        eopatch[out_feature] = np.empty(shape[[0, -1]], dtype=object)
+                        for k in range(shape[-1]):
+                            eopatch[out_feature][time_, k] = td.TDigest()
+                            eopatch[out_feature][time_, k].batch_update(eopatch[in_feature][time_, ..., k].flatten())
+
+        # monthly mode
+        elif self.mode == "monthly":
+            midx = []
+            for month_ in range(12):
+                midx.append(np.array([timestamp.month == month_ + 1 for timestamp in eopatch["timestamp"]]))
+
+            for in_feature, out_feature, pixelwise in zip(self.in_feature, self.out_feature, self.pixelwise):
+                shape = np.array(eopatch[in_feature].shape)
+                if pixelwise:
+                    eopatch[out_feature] = np.empty([12, *shape[1:]], dtype=object)
+                    if len(shape) == 4:  # add feature is temporal check
+                        for month_, i, j, k in product(range(12), range(shape[1]), range(shape[2]), range(shape[3])):
+                            eopatch[out_feature][month_, i, j, k] = td.TDigest()
+                            eopatch[out_feature][month_, i, j, k].batch_update(
+                                eopatch[in_feature][midx[month_], i, j, k].flatten()
+                            )
+                    else:
+                        raise RuntimeError(
+                            f"TDigestTask: {self.mode} mode only defined for DATA or MASK feature types with time"
+                            " component!"
+                        )
+
+                else:
+                    eopatch[out_feature] = np.empty([12, shape[-1]], dtype=object)
+                    for month_, k in product(range(12), range(shape[-1])):
+                        eopatch[out_feature][month_, k] = td.TDigest()
+                        eopatch[out_feature][month_, k].batch_update(
+                            eopatch[in_feature][midx[month_], ..., k].flatten()
+                        )
+
+        # to be implemented modes
+        elif self.mode == "weekly" or self.mode == "daily":
+            raise NotImplementedError(f"TDigestTask: {self.mode} mode not implemented yet!")
+
+        # total mode
+        elif self.mode == "total":
+            for in_feature, out_feature, pixelwise in zip(self.in_feature, self.out_feature, self.pixelwise):
+                if pixelwise:
+                    raise NotImplementedError(f"TDigestTask: pixelwise for {self.mode} mode not implemented yet!")
+
+                else:
+                    eopatch[out_feature] = td.TDigest()
+                    eopatch[out_feature].batch_update(eopatch[in_feature].flatten())
+
+        # errorneous modes
+        else:
+            raise RuntimeError(f"TDigestTask: mode {self.mode} not implemented!")
```

Comment:
anything that is not implemented should be removed from the code. Since some `mode` types are not supported, they should also be removed from the docs. It is simple to add them in a future MR if needed, no need for such placeholders.

For the `total + pixelwise` issue, the error should be raised in the `init` and should probably be a `ValueError`. Also mention in the docs that the combination is not possible.